postgres/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM python:3.8-slim
+
+# Add the path for poetry
+ENV PATH $PATH:/root/.local/bin/
+
+RUN apt-get update && apt-get install -y curl && \
+    # Install & Configure poetry
+    python3 -m pip install --upgrade pip && \
+    curl -sSL https://install.python-poetry.org | python3 - --version 1.4.2 && \
+    poetry config virtualenvs.create false
+
+COPY poetry.lock pyproject.toml /app/
+COPY ./src /app/src
+WORKDIR /app
+RUN --mount=type=cache,target=/home/.cache/pypoetry/cache \
+    --mount=type=cache,target=/home/.cache/pypoetry/artifacts \ 
+    poetry install --only main
+ENV PYTHONPATH /app
+
+# Expose the port on which the application will run (default: 8000)
+EXPOSE 8000
+
+# Run the FastAPI application with Uvicorn
+CMD ["uvicorn", "src.app:app", "--host", "0.0.0.0", "--port", "8000"]
```

Comment:
I'd rather put this command in the docker-compose file. (see: comment on compose file).

It provides more flexibility. You can easily change the command or its arguments without rebuilding the image. And you can reuse the same built image in different context (if needed)